client/app/services/blueprints-state.service.js

```diff
@@ -78,6 +78,57 @@
       return blueprint.doNotSaveFlag;
     };
 
+    blueprint.publishBlueprint = function(tmpBlueprint) {
+      var deferred = $q.defer();
+      var origBlueprint = blueprint.getOriginalBlueprint();
+      if (!angular.equals(origBlueprint, tmpBlueprint)) {
+        // save the changed blueprint
+        blueprint.saveBlueprint(tmpBlueprint).then(function(id) {
+          blueprint.publishBlueprintOp(tmpBlueprint).then(function() {
+            deferred.resolve();
+          }, publishfailure);
+        }, savefailure);
+      } else {
+        // just publish the unchanged blueprint
+        blueprint.publishBlueprintOp(tmpBlueprint).then(function() {
```

Comment:
yeah yeah!! up to you... though maybe i'm using too liberally, most feedback i'm providing is to more align voice of pr with voice of existing codebase

to the second point totally agree!  it doesn't always make sense, but most of time ya can get away with the simple implementations